### PR TITLE
Add Show[List]

### PR DESCRIPTION
**Pull request by ceedubs**

### CODE REVIEW

File: core/src/main/scala/cats/std/list.scala [hunk 1]
```diff
@@ -70,6 +71,11 @@ trait ListInstances extends ListInstances1 {
 
   implicit def listAlgebra[A]: Monoid[List[A]] = new ListMonoid[A]
   implicit def listOrder[A: Order]: Order[List[A]] = new ListOrder[A]
+
+  implicit def listShow[A:Show]: Show[List[A]] =
+    new Show[List[A]] {
+      def show(fa: List[A]): String = s"List(${fa.map(_.show).mkString(", ")})"
```
Comment (fthomas) [inline core/src/main/scala/cats/std/list.scala]: If you're already using `mkString`: `fa.map(_.show).mkString("List(", ", ", ")")`

Comment (ceedubs) [inline core/src/main/scala/cats/std/list.scala]: Oh I forgot you could do that. Thanks will do.

Comment (ceedubs) [inline core/src/main/scala/cats/std/list.scala]: To save you a comment: I'll do that in the Set PR too :)

**Comment by mpilquist** [comment]
:+1:

**Comment by adelbertc** [comment]
:+1:

**Pull request merged**